### ISSUE TITLE
can not find the data correctly when use DDP train

### ISSUE THREAD

**Issue by programmerleel** [ultralytics/ultralytics#14094]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

when i train wiht the follow code:
import os
from ultralytics.models.yolo.model import YOLO

if __name__ == '__main__':
    model = YOLO(r"yolov8l.pt")  # pretrained YOLOv8n model

    model.train(
        data="phone.yaml",
        epochs=100,
        batch=8,
        imgsz=640,
        workers=2,
        project="phone",
        name="2024_6_28_yolov8l",
        device=[0,1,2,3]
    )
it shows  0 images, 7201 backgrounds, 0 corrupt and can not find the labels,but when i use the code without "device=[0,1,2,3]", everything is fine

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@programmerleel hi there,

Thank you for reaching out and providing details about the issue you're encountering with DDP (Distributed Data Parallel) training.

To better assist you, it would be very helpful if you could provide a minimal reproducible example. This will allow us to understand the context and replicate the issue on our end. You can find guidelines on how to create a reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

In the meantime, please ensure that you are using the latest versions of all relevant packages, as updates often include important bug fixes and improvements.

From your description, it seems like the issue might be related to how the data is being accessed or distributed across multiple devices. Here are a few steps you can take to troubleshoot:

1. **Verify Data Paths**: Ensure that the data paths specified in your `phone.yaml` file are accessible from all devices. Sometimes, relative paths can cause issues in a distributed setup.

2. **Check Data Loading**: When using multiple devices, the data loading process can sometimes behave differently. You might want to increase the number of workers or check if the data is being correctly distributed across the devices.

3. **Environment Variables**: Ensure that the environment variables for DDP are correctly set. Sometimes, issues can arise from improper setup of the distributed environment.

Here's a small code snippet to help you verify the data loading process:

```python
import os
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO(r"yolov8l.pt")  # pretrained YOLOv8n model

    # Verify data loading
    data = model.data.load('phone.yaml')
    print(f"Training data: {data}")

    # Train the model
    model.train(
        data="phone.yaml",
        epochs=100,
        batch=8,
        imgsz=640,
        workers=2,
        project="phone",
        name="2024_6_28_yolov8l",
        device=[0,1,2,3]
    )
```

This will help ensure that the data is being correctly loaded before the training process begins.

If the issue persists, please provide additional details or a reproducible example, and we will be happy to assist you further.
